### PR TITLE
[dynamo] allow extending classes in `make_test_cls_with_patches`

### PR DESCRIPTION
**Pull request by nkaretnikov**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #91952
* __->__ #92228

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

**Comment by nkaretnikov** [comment]
Extended this function to reuse it in the next commit in the stack. This is a separate PR to make it easier to review.

File: torch/_dynamo/testing.py [hunk 1]
```diff
@@ -255,33 +256,49 @@ def rand_strided(size, stride, dtype=torch.float32, device="cpu", extra_size=0):
...
+    cls: Any,  # class object
+    make_new_cls: bool,  # or add to existing class
+    new_fn_suffix: str,
+    new_cls_prefix: Optional[str] = None,  # if make_new_cls
+    name_pred: Callable[[str], bool] = lambda name: name.startswith("test_"),
+    patches=Tuple[Tuple[Any, str, Any]],  # module, attr, val
+):
+    if make_new_cls:
 
-    DummyTestClass.__name__ = f"{cls_prefix}{cls.__name__}"
```
Comment (nkaretnikov) [inline torch/_dynamo/testing.py:274]: the newline above might look strange, but it's required by the linter

File: test/dynamo/test_dynamic_shapes.py [hunk 2]
```diff
@@ -26,7 +27,11 @@
 
 def make_dynamic_cls(cls):
     return make_test_cls_with_patches(
-        cls, "DynamicShapes", "_dynamic_shapes", ("dynamic_shapes", True)
+        cls=cls,
+        make_new_cls=True,
```
Comment (ezyang) [inline test/dynamo/test_dynamic_shapes.py:31]: when would `make_new_cls` ever be false 🤔

Comment (nkaretnikov) [reply test/dynamo/test_dynamic_shapes.py:31]: It was done for the other commit in the stack. But now that you want me to generate all tests, let's just do that. Then we won't need to touch this either, so no review overhead.

**Comment by nkaretnikov** [comment]
Decided to go with a different approach for inductor tests, so this one is no longer needed. Closing.

**Pull request closed without merging**